### PR TITLE
Support $brooklyn:external() in YAML location config.

### PR DESCRIPTION
**Pull request by alasdairhodge**

### CODE REVIEW

**Comment by sjcorbett** [comment]
@alasdairhodge Could you rebase on master?

**Comment by alasdairhodge** [comment]
Now rebased, @sjcorbett

File: core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTaskTags.java [hunk 1]
```diff
@@ -157,6 +158,13 @@ public static Entity getTargetOrContextEntity(Task<?> t) {
         return em.getTasksWithTag(tagForContextEntity(e));
     }
 
+    public static ManagementContext getManagementContext(Task<?> task) {
```
Comment (sjcorbett) [inline core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTaskTags.java]: Javadoc about possible return values worthwhile.

**Comment by sjcorbett** [comment]
@alasdairhodge Looks good. I'd like to see tests for the behaviour but will merge now.

**Comment by alasdairhodge** [comment]
Thanks, @sjcorbett.

Re: tests, Alex added one as part of #922; I removed it form the `WIP` group here.

**Pull request merged**